File: functions/src/types/classes.ts
export type Classes = {
  /**
   * 授業があるか
   * items が空の場合 true
   */
  isEmpty: boolean;

  /**
   * 授業リスト
   */
  items: ClassData[];
};

export type ClassData = {
  /**
   * 授業名
   */
  title: string;

  /**
   * 開始時刻
   * ISO 形式
   */
  startAt: string;

  /**
   * 終了時刻
   * ISO 形式
   */
  endAt: string;

  /**
   * 何時間目か
   */
  period: number;
};

File: frontend/lib/models/classes.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:nocsis/converters/local_date_time.dart';

part 'classes.freezed.dart';
part 'classes.g.dart';

@freezed
abstract class ClassList with _$ClassList {
  const factory ClassList({required List<ClassData> items}) = _ClassList;

  factory ClassList.fromJson(Map<String, dynamic> json) =>
      _$ClassListFromJson(json);
}

@freezed
abstract class ClassData with _$ClassData {
  const factory ClassData({
    required String title,
    required int period,
    @LocalDateTimeConverter() required DateTime startAt,
    @LocalDateTimeConverter() required DateTime endAt,
  }) = _ClassData;

  factory ClassData.fromJson(Map<String, dynamic> json) =>
      _$ClassDataFromJson(json);
}
